packages/react-components/src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,177 +1,158 @@
     } else {
-      visible && onClose?.();
-    }
-    if (!isManaged) {
-      setVisibility(newVisibility);
+      onOpen?.();
+      !isControlled && setIsVisible(true);
     }
   };
 
-  React.useEffect(() => {
-    // handles visibility changes from outside the component
-    if (isFirstRender.current) {
-      isFirstRender.current = false;
-
-      return;
+  const getTransitionValue = (value?: number) => {
+    if (!withFadeAnimation) {
+      return 0;
     }
-    if (isVisible === true) onOpen?.();
-    if (isVisible === false) onClose?.(); // we need to check if it's false, because it can be undefined
-
-    setVisibility(isVisible);
-  }, [isVisible]);
-
-  React.useEffect(() => {
-    document.addEventListener('keydown', handleCloseAction);
-
-    return () => {
-      document.removeEventListener('keydown', handleCloseAction);
-    };
-  }, []);
 
-  const handleMouseEnter = () => {
-    if (triggerOnClick || isManaged) return;
-    isHovered.current = true;
-    handleVisibilityChange(true);
+    return value ? value : transitionDuration;
   };
 
-  const handleMouseLeave = () => {
-    if (triggerOnClick || isManaged) return;
-    isHovered.current = false;
-    void sleep(hoverOutDelayTimeout).then(() => {
-      if (!isHovered.current) {
-        handleVisibilityChange(false);
-      }
-    });
+  const getTransitionDelayValue = (value?: number) => {
+    return value ? value : transitionDelay;
   };
 
-  const handleCloseAction = (event: KeyboardEvent | MouseEvent) => {
-    if (
-      (event instanceof KeyboardEvent && event.key === 'Escape') ||
-      event.type === 'click'
-    ) {
-      handleVisibilityChange(false);
-    }
-  };
-
-  const floatingComponent = (
-    <FloatingComponent
-      baseClass={baseClass}
-      className={className}
-      visible={visible}
-      floatingOptions={floatingOptions}
-      arrowRef={arrowRef}
-      handleMouseEnter={handleMouseEnter}
-      handleMouseLeave={handleMouseLeave}
-      handleCloseAction={handleCloseAction}
-      childrenElements={children}
-      transitionDuration={transitionDuration}
-      transitionDelay={transitionDelay}
-      referenceElement={referenceElement}
-      arrowOffsetX={arrowOffsetX}
-      arrowOffsetY={arrowOffsetY}
-      theme={theme}
-      withFadeAnimation={withFadeAnimation}
-      fullSpaceContent={fullSpaceContent}
-    />
-  );
-
-  if (referenceElement) {
-    return floatingComponent;
-  }
-
-  const handleClick = () => {
-    if (visible) {
-      handleVisibilityChange(false);
-    } else {
-      handleVisibilityChange(true);
-    }
-  };
+  const { floatingStyles, refs, context } = useFloating({
+    middleware: [
+      offset({ mainAxis: offsetMainAxis }),
+      shift(),
+      flip(),
+      arrow({ element: arrowRef }),
+    ],
+    placement: placement,
+    open: currentlyVisible,
+    onOpenChange: handleMenuStateChange,
+    whileElementsMounted: autoUpdate,
+  });
+  const hover = useHover(context, {
```

Comment:
add https://floating-ui.com/docs/useHover#restms support and create a corresponding component prop - `activationThreshold` for example